src/shared/components/dupui/file-upload.tsx

```
@@ -0,0 +1,189 @@
+
+  const handleClick = useCallback(() => {
+    fileInputRef.current?.click();
+  }, []);
+
+  // Simple binary content detection
+  const containsBinaryContent = (text: string): boolean => {
+    // Check for null bytes (common in binary files)
+    if (text.includes('\0')) return true;
+    
+    // Check for high ratio of non-printable characters
+    const nonPrintable = text.replace(/[\x20-\x7E\x0A\x0D\x09]/g, '').length;
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)